packages/react/src/state.tsx

```diff
@@ -115,4 +134,32 @@ export const increment = (value: number): IncrementStateModifier => ({
   value,
 })
 
-export const LocalState = createContext<LocalState>({})
+export const useLocalStateValues = () => {
+  const localState = useContext(LocalState)
+
+  const [, forceUpdate] = useState({})
+  const subscriptions = useRef(new Map())
+
+  function get(state: StateDataState) {
+    const stream = localState.getStream(state)
+
+    if (!subscriptions.current.has(state.key)) {
+      const subscription = stream.subscribe(() => {
+        forceUpdate({})
+      })
+      subscriptions.current.set(state.key, subscription)
+    }
+
+    return stream.get()
+  }
+
+  useEffect(() => {
```

Comment:
hmm, one code smell is that usually the subscription will happen in the useEffect that gets cleaned up in the return value of useEffect. like, does this even work with the double effect thing that React does in dev mode?

Reply:
Interesting! I've been mostly testing this in unit tests where this does not apply of course. Good catch, we need to handle it!

Reply:
Fixed in https://github.com/final-ui/final/pull/76/commits/3d0f4f972478972ab697752c928006d1bd8bca8b